File: Fetch stations/FetchStations/Networking/Station.m
//
//  Station.m
//  FetchStations
//
//  Created by Waylan Sands on 3/5/2023.
//

#import <Foundation/Foundation.h>
#import "Station.h"

@implementation Station

- (instancetype)initWithPlace:(Place *)place {
    self = [super init];
    if (self) {
        self.place = place;
    }
    return self;
}

@end

File: FetchStations/Models/Station.swift
//
//  Station.swift
//  FetchStations
//
//  Created by Waylan Sands on 3/5/2023.
//

import Foundation

struct StationResponse: Codable {
    let stations: [Station]
}

struct Station: Codable {
    let place: Place
}

struct Place: Codable {
    let location: Location
    let name: String
    let type: String
    let id: String
}

struct Location: Codable {
    let lat: Double
    let lng: Double
}
